reconcile/terraform_tgw_attachments.py

```diff
@@ -471,7 +471,9 @@ def run(
 
 
 def early_exit_desired_state(*args: Any, **kwargs: Any) -> dict[str, Any]:
-    return _fetch_desired_state_data_source().dict(by_alias=True)
+    datasource = _fetch_desired_state_data_source()
+    datasource.accounts = _filter_tgw_accounts(datasource.accounts, datasource.clusters)
```

Comment:
can we do this inside `_fetch_desired_state_data_source`, are non filtered accounts used anywhere?

Reply:
or split `accounts` into `all_accounts` and `tgw_accounts` so the same `DesiredStateDataSource` can be used directly.

Reply:
> are non filtered accounts used anywhere?
yes they are now. They're used to get cluster accounts (non-tgw).

> split accounts into all_accounts and tgw_accounts so the same DesiredStateDataSource can be used directly.
But if `all_account` is in the desired state `datasource.dict(by_alias=True)` will also return all accounts in the early exit. Or do you then propose to only return `clusters` and `tgw_accounts` ? It looks pretty much the same than what is proposed here IMHO.

Reply:
If `all_account` is used, should it be included in early exit state anyway? Otherwise it may wrongly exit.

Reply:
I think it should not. From `all_accounts` we end up using only the ones with TGW and the ones corresponding to ROSA/CCS clusters

Reply:
if ehe ones corresponding to ROSA/CCS clusters changed, will desired state change?

Reply:
So what we need is `tgw_rosa_cluster_accounts` and `tgw_accounts`, right? Are we using any other accounts inside `all_accounts`?

Reply:
> if the ones corresponding to ROSA/CCS clusters changed, will desired state change?

No

> So what we need is tgw_rosa_cluster_accounts and tgw_accounts, right? 

I don't understand what you want to gain there.